tests/store/artifact/test_s3_artifact_repo.py

```diff
@@ -218,3 +218,31 @@ def test_get_s3_file_upload_extra_args_invalid_json():
 
     with pytest.raises(ValueError):
         S3ArtifactRepository.get_s3_file_upload_extra_args()
+
+
+def test_delete_artifacts(s3_artifact_root, tmpdir):
+    subdir_path = str(tmpdir.mkdir("subdir"))
+    nested_path = os.path.join(subdir_path, "nested")
+    os.makedirs(nested_path)
+    path_a = os.path.join(subdir_path, "a.txt")
+    path_b = os.path.join(subdir_path, "b.tar.gz")
+    path_c = os.path.join(nested_path, "c.csv")
+
+    with open(path_a, "w") as f:
+        f.write("A")
+    with tarfile.open(path_b, "w:gz") as f:
+        f.add(path_a)
+    with open(path_c, "w") as f:
+        f.write("col1,col2\n1,3\n2,4\n")
+
+    repo = get_artifact_repository(posixpath.join(s3_artifact_root, "some/path"))
+    repo.log_artifacts(subdir_path)
+
+    repo.delete_artifacts()
+
+    tmpdir_objects = repo.list_artifacts()
```

Comment:
I think the artifacts list should be empty after deletion, right?

Also, the `path` attributes of artifact info returned by `list_artifacts` refer to run-relative artifact paths (e.g., `"a.txt"` rather than `path_a`. (See https://mlflow.org/docs/latest/rest-api.html#fileinfo)

Reply:
> I think the artifacts list should be empty after deletion, right?
Again, right :). Just checking that the list is empty should be enough, instead of 3 asserts. 

I will provide an update soon.